crates/l2/sequencer/l1_proof_sender.rs

```diff
@@ -101,118 +96,161 @@ impl L1ProofSender {
             proof_send_interval_ms: cfg.proof_send_interval_ms,
         })
     }
+}
 
-    async fn run(&self) {
-        loop {
-            info!("Running L1 Proof Sender");
-            info!("Needed proof systems: {:?}", self.needed_proof_types);
-            let _ = self
-                .main_logic()
-                .await
-                .inspect_err(|err| error!("L1 Proof Sender Error: {err}"));
+#[derive(Clone)]
+pub enum InMessage {
+    Send,
+}
 
-            sleep_random(self.proof_send_interval_ms).await;
-        }
+#[allow(dead_code)]
+#[derive(Clone, PartialEq)]
+pub enum OutMessage {
+    Done,
+    Error,
+}
+
+pub struct L1ProofSender;
+
+impl L1ProofSender {
+    pub async fn spawn(cfg: SequencerConfig) -> Result<(), ProofSenderError> {
+        let state =
+            L1ProofSenderState::new(&cfg.proof_coordinator, &cfg.l1_committer, &cfg.eth).await?;
+        let mut l1_proof_sender = L1ProofSender::start(state);
+        l1_proof_sender
+            .cast(InMessage::Send)
+            .await
+            .map_err(ProofSenderError::GenServerError)?;
```

Comment:
Nit. This `map_err` is not needed when the target error (`ProofSenderError`) has a variant that derives the local error (`ProofSenderError::GenServerError` derives from`spawned_concurrency::GenServerError `).
```suggestion
            .await?;
```